pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/UnusedAssignment.xml

```diff
@@ -1240,6 +1296,68 @@ public class Foo {
 
 }        ]]></code>
     </test-code>
+    <test-code>
+        <description>Try/catch finally 4</description>
+        <expected-problems>3</expected-problems>
+        <expected-linenumbers>4,6,8</expected-linenumbers>
+        <expected-messages>
+            <message>The initializer for variable 'a' is never used (overwritten on lines 6, 8 and 10)</message>
+            <message>The value assigned to variable 'a' is never used (overwritten on line 10)</message>
+            <message>The value assigned to variable 'a' is never used (overwritten on line 10)</message>
+        </expected-messages>
+        <code><![CDATA[
+public class Foo {
+
+    public int foo() {
+        int a = 0; // overwritten in try body & catch
+        try (Reader r = new StringReader("")) {
```

Comment:
Theoretically, creating the reader could already throw (e.g. when using FileReader), which means, that catch & finally is directly executed, but not the try-body.
But in this example, a is always overwritten in finally, so the result doesn't change.

Reply:
I think this should be handled.. I'll double check later